utils/records-decorator.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var _ = require('lodash');
+
+function decorateForSearch(records, fieldsSearched, searchValue) {
+  var matchFields = {};
+  records.forEach(function (record, index) {
+    Object.keys(record.rawAttributes).forEach(function (attributeName) {
+      var value = record[attributeName];
+      if (value) {
+        value = value.toString();
+        var match = fieldsSearched.includes(attributeName) && value.match(new RegExp(searchValue, 'i'));
+        if (match) {
+          if (!matchFields[index]) {
+            matchFields[index] = {
+              id: record.id,
+              search: [],
+            };
+          }
+          matchFields[index]['search'].push(attributeName);
+        }
+      }
+    });
+  });
+
+  if (_.isEmpty(matchFields)) {
+    matchFields = null;
+  }
+
+  return matchFields;
+}
+
+exports.decorateForSearch = decorateForSearch;
```

Comment:
This service as to be exposed to the `forest-express` liana to make it run just before the serialization.
The problem with the current implementation is that Smart Fields are not highlighted if they contain the search value. And Smart Field values injection is done in the `forest-express` liana...